src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -580,6 +629,34 @@ internal FunctionPointerMethodSymbol ApplyNullableTransforms(byte defaultTransfo
             }
         }
 
+        internal override ImmutableArray<INamedTypeSymbol> CallingConventionTypes
+        {
+            get
+            {
+                if (!CallingConvention.IsCallingConvention(CallingConvention.Unmanaged))
+                {
+                    return ImmutableArray<INamedTypeSymbol>.Empty;
+                }
+
+                var modifiersToSearch = RefKind != RefKind.None ? RefCustomModifiers : ReturnTypeWithAnnotations.CustomModifiers;
```

Comment:
>var modifiersToSearch = RefKind != RefKind.None ? RefCustomModifiers : ReturnTypeWithAnnotations.CustomModifiers; [](start = 16, length = 113)

It feels like there is a fair amount of code duplication here. Should we simply delegate most of the work to GetCallingConventionModifiers?

Reply:
I can make a helper for getting the correct set of modifiers to search, but this version specifically makes sure that order and duplication are preserved so that the IDE can generate code that matches what the user already had, so I don't think the actual search and population logic can be preserved.

---
In reply to: [463859015](https://github.com/dotnet/roslyn/pull/46467#discussion_r463859015) [](ancestors = 463859015)

Reply:
I see, the order and duplicate significance should be clearly documented then.

---
In reply to: [463871734](https://github.com/dotnet/roslyn/pull/46467#discussion_r463871734) [](ancestors = 463871734,463859015)